include/nbl/core/alloc/VectorViewNullMemoryResource.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2019-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine" and was originally part of the "Irrlicht Engine"
+// For conditions of distribution and use, see copyright notice in nabla.h
+// See the original file in irrlicht source for authors
+
+#ifndef _NBL_CORE_ALLOC_VECTOR_VIEW_NULL_MEMORY_RESOURCE_INCLUDED_
+#define _NBL_CORE_ALLOC_VECTOR_VIEW_NULL_MEMORY_RESOURCE_INCLUDED_
+
+#include <memory_resource>
+
+using namespace nbl;
+
+namespace nbl::core
+{
+
+class VectorViewNullMemoryResource : public std::pmr::memory_resource
+{
+    public:
+        // only create the resource from an already sized vector
+        VectorViewNullMemoryResource(core::vector<uint8_t>&& buffer) : buffer(buffer)
+        {
+            assert(buffer.size());
+        };
+
+        void* data() {
+            return buffer.data();
+        }
+
+    protected:
+        void* do_allocate(size_t bytes, size_t alignment) override {
+            assert(bytes <= buffer.size());
+            return buffer.data();
```

Comment:
if you want you can have a `bool` sentinel value to check that `do_allocate` was called onceand only once

Reply:
actually you should probably return `nullptr` instead of `assert`-ing, also return nullptr if `buffer.data()` is `!core::is_aligned_to` the `alignment